### ISSUE TITLE
Can’t use with Prowlarr and Yggtorrent

### ISSUE THREAD

**Issue by bajire72** [FlareSolverr/FlareSolverr#705]
### Have you checked our README?

- [X] I have checked the README

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you checked the discussions?

- [X] I have read the Discussions

### Environment

```markdown
- FlareSolverr version: 3.0.2
- Last working FlareSolverr version: 3.0.2
- Operating system: Windows
- Are you using Docker: [yes/no] yes
- FlareSolverr User-Agent (see log traces or / endpoint): Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36
- Are you using a VPN: [yes/no] No
- Are you using a Proxy: [yes/no] No
- Are you using Captcha Solver: [yes/no] No
- If using captcha solver, which one:
- URL to test this issue:
```

### Description

Hello, FlareSolverr doesn’t work anymore with Yggtorrent do you have any solution ?
There is my Prowlarr logs

### Logged Error Messages

```text
[Warn] ProwlarrErrorPipeline: FluentValidation.ValidationException: Validation failed: 
 -- : Unable to connect to indexer, check the log for more details 

[v1.1.3.2521] FluentValidation.ValidationException: Validation failed: 
 -- : Unable to connect to indexer, check the log for more details
   at Prowlarr.Api.V1.ProviderControllerBase`3.VerifyValidationResult(ValidationResult validationResult, Boolean includeWarnings) in D:\a\1\s\src\Prowlarr.Api.V1\ProviderControllerBase.cs:line 209
   at Prowlarr.Api.V1.ProviderControllerBase`3.Test(TProviderDefinition definition, Boolean includeWarnings) in D:\a\1\s\src\Prowlarr.Api.V1\ProviderControllerBase.cs:line 195
   at Prowlarr.Api.V1.ProviderControllerBase`3.UpdateProvider(TProviderResource providerResource) in D:\a\1\s\src\Prowlarr.Api.V1\ProviderControllerBase.cs:line 87
   at lambda_method122(Closure , Object , Object[] )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Prowlarr.Http.Middleware.BufferingMiddleware.InvokeAsync(HttpContext context) in D:\a\1\s\src\Prowlarr.Http\Middleware\BufferingMiddleware.cs:line 28
   at Prowlarr.Http.Middleware.IfModifiedMiddleware.InvokeAsync(HttpContext context) in D:\a\1\s\src\Prowlarr.Http\Middleware\IfModifiedMiddleware.cs:line 41
   at Prowlarr.Http.Middleware.CacheHeaderMiddleware.InvokeAsync(HttpContext context) in D:\a\1\s\src\Prowlarr.Http\Middleware\CacheHeaderMiddleware.cs:line 33
   at Prowlarr.Http.Middleware.UrlBaseMiddleware.InvokeAsync(HttpContext context) in D:\a\1\s\src\Prowlarr.Http\Middleware\UrlBaseMiddleware.cs:line 27
   at Prowlarr.Http.Middleware.VersionMiddleware.InvokeAsync(HttpContext context) in D:\a\1\s\src\Prowlarr.Http\Middleware\VersionMiddleware.cs:line 28
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)
```

**Comment by markettepigeon** [comment]
Hello

Same problem for YggTorrent :

An error occurred while testing this indexer
Exception (yggtorrent): FlareSolverr was unable to process the request, please check FlareSolverr logs. Message: Error: Error solving the challenge. Timeout after 55.0 seconds.: FlareSolverr was unable to process the request, please check FlareSolverr logs. Message: Error: Error solving the challenge. Timeout after 55.0 seconds

**Comment by TheNexter** [comment]
> Hello
> 
> Same problem for YggTorrent :
> 
> An error occurred while testing this indexer Exception (yggtorrent): FlareSolverr was unable to process the request, please check FlareSolverr logs. Message: Error: Error solving the challenge. Timeout after 55.0 seconds.: FlareSolverr was unable to process the request, please check FlareSolverr logs. Message: Error: Error solving the challenge. Timeout after 55.0 seconds

Hello, same here (latest update)

**Comment by Write** [comment]
Yup, need to use Jackett for now.

**Comment by Thomas-Philippot** [comment]
yes, unable to use Flaresolverr even on a machin that can access yggtorrent on the browser

**Comment by bajire72** [comment]
> Yup, need to use Jackett for now.
> **EDIT:** Nevermind, it actually does the same error, undetected-chromedriver seems detected by cloudflare now.

Yeah doesn’t work …

**Comment by bajire72** [comment]
> Hello
> 
> Same problem for YggTorrent :
> 
> An error occurred while testing this indexer Exception (yggtorrent): FlareSolverr was unable to process the request, please check FlareSolverr logs. Message: Error: Error solving the challenge. Timeout after 55.0 seconds.: FlareSolverr was unable to process the request, please check FlareSolverr logs. Message: Error: Error solving the challenge. Timeout after 55.0 seconds

This is my error code with Jackett

**Comment by charliik** [comment]
Hello, same here, not working either with Prowlarr or Jackett: 

`Exception (yggtorrent): The cookies provided by FlareSolverr are not valid: The cookies provided by FlareSolverr are not valid`

**Comment by vyannis** [comment]
Same here, same error ...

**Comment by JubeiQC** [comment]
Same here, Prowlarr and Jackett

**Comment by mistertix** [comment]
same here

**Comment by Guesh1337** [comment]
same here on jackett

**Comment by GregoireDelorme** [comment]
Not sure it''s fully related to the undetected chrome driver : i got the same issue on mobile using my VPN: Probably linked to the ASN / IP  

#694  Currently opened . I see no other solution than tryng to solve the basic captcha ( and it will be probably temporary until their detection give a more difficult one) or use proxies with a more trusted ip

**Comment by Brainwasher06** [comment]
Same here, on both Prowlarr and Jackett

**Comment by iwa** [comment]
no longer works with yggcookies :/

**Comment by ilike2burnthing** [comment]
If you're using Jackett/Prowlarr and FlareSolverr on the same device and the same IP (i.e. no VPN or proxy), with IPv6 disabled, and other indexers which require FlareSolverr are working (TorrentQQ is a quick test), then the issue lies with YGG.

There are plenty of old issues regarding YGG, and they almost all come down to the same thing: their implementation of Cloudflare IAUM protection as a result of them being DDoS'd. There's nothing we can do about that.

**Comment by iwa** [comment]
@ilike2burnthing 

I made the test, I have prowlarr with flaresolverr, both running through docker, ipv4 only, no vpn, and even when I'm trying to add TorrentQQ, it's throwing me a timeout:

```
2023-02-18 12:15:36 INFO     Incoming request => POST /v1 body: {'maxTimeout': 60000, 'cmd': 'request.get', 'url': 'https://torrentqq244.com/torrent/newest.html', 'userAgent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36', 'proxy': {}}
2023-02-18 12:15:36 WARNING  Request parameter 'userAgent' was removed in FlareSolverr v2.
2023-02-18 12:15:37 INFO     Challenge detected. Title found: Just a moment...
2023-02-18 12:16:37 ERROR    Error: Error solving the challenge. Timeout after 60.0 seconds.
2023-02-18 12:16:37 INFO     Response in 60.652 s
2023-02-18 12:16:37 INFO     192.168.160.4 POST http://flaresolverr:8191/v1 500 Internal Server Error
```

So even with the junky implementation of ygg's cloudflare protection, are we really sure the flaresolverr web agent is still undetected?

**Comment by Write** [comment]
> @ilike2burnthing
> 
> I made the test, I have prowlarr with flaresolverr, both running through docker, ipv4 only, no vpn, and even when I'm trying to add TorrentQQ, it's throwing me a timeout:
> 
> ```
> 2023-02-18 12:15:36 INFO     Incoming request => POST /v1 body: {'maxTimeout': 60000, 'cmd': 'request.get', 'url': 'https://torrentqq244.com/torrent/newest.html', 'userAgent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36', 'proxy': {}}
> 2023-02-18 12:15:36 WARNING  Request parameter 'userAgent' was removed in FlareSolverr v2.
> 2023-02-18 12:15:37 INFO     Challenge detected. Title found: Just a moment...
> 2023-02-18 12:16:37 ERROR    Error: Error solving the challenge. Timeout after 60.0 seconds.
> 2023-02-18 12:16:37 INFO     Response in 60.652 s
> 2023-02-18 12:16:37 INFO     192.168.160.4 POST http://flaresolverr:8191/v1 500 Internal Server Error
> ```
> 
> So even with the junky implementation of ygg's cloudflare protection, are we really sure the flaresolverr web agent is still undetected?

Seems undetected still, It does exactly the same thing when I'm just with the same VPN IP as my Server but on my regular PC. Sometimes it show the blue button to click or sometimes the checkmark to click, but in both case it's random. Both with my script undetected or my regular browser

**Comment by ilike2burnthing** [comment]
@iwa if you're using a residential IP (so not a seedbox or VPS), then there's some wider issue for you. Try increasing the timeout.

**Comment by Bastien42** [comment]
```
2023-02-18 13:12:35 2023-02-18 12:12:35 WARNING  ReqId 139990408668928 Request parameter 'userAgent' was removed in FlareSolverr v2.
2023-02-18 13:12:35 2023-02-18 12:12:35 DEBUG    ReqId 139990408668928 Launching web browser...
2023-02-18 13:12:35 2023-02-18 12:12:35 DEBUG    ReqId 139990408668928 Started executable: `/app/chromedriver` in a child process with pid: 167
2023-02-18 13:12:36 2023-02-18 12:12:36 DEBUG    ReqId 139990375098112 Navigating to... https://www6.yggtorrent.lol/engine/search?category=2145&name=&description=&file=&uploader=&sub_category=&do=search&order=desc&sort=publish_date
2023-02-18 13:12:36 2023-02-18 12:12:36 INFO     ReqId 139990375098112 Challenge detected. Title found: Just a moment...
2023-02-18 13:12:36 2023-02-18 12:12:36 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:12:46 2023-02-18 12:12:46 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:12:46 2023-02-18 12:12:46 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:12:57 2023-02-18 12:12:57 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:12:57 2023-02-18 12:12:57 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:13:07 2023-02-18 12:13:07 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:13:07 2023-02-18 12:13:07 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:13:17 2023-02-18 12:13:17 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:13:17 2023-02-18 12:13:17 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:13:27 2023-02-18 12:13:27 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:13:27 2023-02-18 12:13:27 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:13:37 2023-02-18 12:13:37 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:13:37 2023-02-18 12:13:37 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:13:47 2023-02-18 12:13:47 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:13:47 2023-02-18 12:13:47 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:13:57 2023-02-18 12:13:57 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:13:57 2023-02-18 12:13:57 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:14:07 2023-02-18 12:14:07 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:14:07 2023-02-18 12:14:07 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:14:17 2023-02-18 12:14:17 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:14:17 2023-02-18 12:14:17 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:14:27 2023-02-18 12:14:27 DEBUG    ReqId 139990375098112 Timeout waiting for selector
2023-02-18 13:14:27 2023-02-18 12:14:27 DEBUG    ReqId 139990375098112 Waiting for title: Just a moment...
2023-02-18 13:14:36 2023-02-18 12:14:36 ERROR    ReqId 139990408668928 Error: Error solving the challenge. Timeout after 120.0 seconds.
2023-02-18 13:14:36 2023-02-18 12:14:36 DEBUG    ReqId 139990408668928 Response => POST /v1 body: {'status': 'error', 'message': 'Error: Error solving the challenge. Timeout after 120.0 seconds.', 'startTimestamp': 1676722355587, 'endTimestamp': 1676722476303, 'version': '3.0.2'}
2023-02-18 13:14:36 2023-02-18 12:14:36 INFO     ReqId 139990408668928 Response in 120.716 s
2023-02-18 13:14:36 2023-02-18 12:14:36 INFO     ReqId 139990408668928 172.17.0.1 POST http://192.168.1.197:8191/v1 500 Internal Server Error
```

residential ip here 120sec TO same issue

**Comment by ilike2burnthing** [comment]
That was specifically for @iwa as they're having issues with TorrentQQ as well.

**Comment by iwa** [comment]
> @iwa if you're using a residential IP (so not a seedbox or VPS), then there's some wider issue for you. Try increasing the timeout.

ah yeah it seems to be working now, sry then that was on me!

**Comment by Thomas-Philippot** [comment]
> > @iwa if you're using a residential IP (so not a seedbox or VPS), then there's some wider issue for you. Try increasing the timeout.
> 
> 
> 
> ah yeah it seems to be working now, sry then that was on me!

Same for me, seems to be working again

**Comment by Bastien42** [comment]
It's back working only because ygg admin lowered the cloudflare protection

**Comment by Jamy50** [comment]
All it's ok since sunday !

**Comment by Write** [comment]
> All it's ok since sunday !

Yes but only because of what @Bastien42 said.

**Comment by Jamy50** [comment]
> > All it's ok since sunday !
> 
> Yes but only because of what @Bastien42 said.

CQFD

**Comment by Compr0mzd** [comment]
Not anymore :)

**Comment by Compr0mzd** [comment]
> Not anymore :)

[Warn] ProwlarrErrorPipeline: Invalid request Validation failed: 
 -- : Unable to access www3.yggtorrent.wtf, blocked by CloudFlare Protection.

**Comment by ilike2burnthing** [comment]
@Compr0mzd make sure the indexer has the FlareSolverr tag.

**Comment by Compr0mzd** [comment]
> @Compr0mzd make sure the indexer has the FlareSolverr tag.

I did

**Comment by ilike2burnthing** [comment]
Open a new issue, providing the information requested and debug FlareSolverr logs.
